streaming/stream_session.cc

```diff
@@ -132,8 +133,11 @@ void stream_manager::init_messaging_service_handler() {
             (void)mutation_writer::distribute_reader_and_consume_on_shards(s,
                 make_generating_reader_v1(s, permit, std::move(get_next_mutation_fragment)),
                 make_streaming_consumer("streaming", _db, _view_update_generator, estimated_partitions, is_offstrategy_supported(reason),
-                        [&sys_dist_ks = _sys_dist_ks, reason] (const replica::table& t) {
-                    return db::view::check_streaming_needs_view_update_path(sys_dist_ks.local(), t, reason);
+                        [&sys_dist_ks = _sys_dist_ks, &g = _gossiper, reason] (const replica::table& t) {
+                    return db::view::check_streaming_needs_view_update_path(sys_dist_ks.local(), t, reason, [&g = g.local()] (locator::host_id host) {
+                        auto nodes = g.get_nodes_with_host_id(host);
+                        return !nodes.empty() && g.is_normal_ring_member(*nodes.begin());
```

Comment:
@bhalevy, @asias is this the best way to filter out uknown hosts?

Reply:
Botond, since we have host_id. I think std::optional<inet_address> get_endpoint_for_host_id(host_id) from token_metadata is the best thing to check. 1) No need to bring gossip in 2) get_endpoint_for_host_id directly accesses the truth of source _endpoint_to_host_id_map for host_ids that are part of the ring (cluster).

I think we can just pass the token_metadata to view code. It is the job of the check_needs_view_update_path to decide how to filter the invalid entries. (Image the view does not store host_id). The caller does not know how to filter. 

If we pass a filter function from the caller, we need to repeat the same filter logic for all make_streaming_consumer caller (repair/row_level.cc and  and streaming/stream_session.cc)